docs/porting/connectivity/Layer3IP.md

```diff
@@ -0,0 +1,144 @@
+    	return l3ip_test_driver;
+	}
+
+	// Weak so a module can override
+	MBED_WEAK L3IP &L3IP::get_default_instance()
+	{
+    	return Cellular_driver_L3IP::get_instance();
+	}
+
+Following driver class example can be found in mbed-os/TESTS/network/l3ip directory.
+
+
+### Being the default Interface
+Currently EMAC based  interfaces are created by automated factories and can be default for applications.
+
+	NetworkInterface::get_default_instance()
+
+Can create ethernet, wifi, mesh or cellular depending on JSON config setting:
+
+    NETWORK_DEFAULT_INTERFACE_TYPE
+
+Interface created that way is set default. 
+
+IP Layer3  Interface is designed to  not use `NetworkInterface::get_default_instance` so it is set as non default on create.
+
+To set IP Layer3  as default interface new member of  `set_as_default()` can be used. This metod is not limited to IP Layer3 and can set any network interface as default.
+
+
+### Cellular interfaces
+
+As a Cellular interface, a little more work is necessary - at a minimum, you need to implement the extra configuration calls in `L3IPInterface`. This is because the network stacks and IP Layer3 APIs only relate to the basic data path - they have no knowledge of any other configuration mechanisms and assume they are already set up.
+
+To do this, create a C++ class that inherits from both `L3IPInterface` and base class for new IP Layer3 cellular interface solution. The `L3IPInterface` is a helper class that implements all the core `NetworkInterface` functionality for you. You then need to implement the extra configuration methods.
+
+
+You don't usually directly expose the `L3IP` driver class of a cellular  driver - it is not usually declared as `L3IP::get_default_instance`, but you would pass it to the constructor of your base `L3IPInterface`. This then makes it visible using the `getl3ip` method. 
+
+
+### OnboardNetworkStack
+
+The precise details of the `OnboardNetworkStack` API should not concern an L3IP driver writer - it provides the mechanism to bind a driver to a stack, and the APIs needed to implement a `NetworkInterface`, but this is handled  by `L3IPInterface`, either as a base class of your own `XXXInterface`.
+
+### Tuning memory allocations
+
+Depending on its use of pool and heap memory and other factors, a driver might want to tune the configuration of particular network stacks. You can do this using the `mbed_lib.json` of each network stack, using their `target_overrides` section.
+
+### Testing
+
+The Mbed OS tree contains Greentea-based tests that exercise the L3IP API directly, and more general socket tests.
+
+For general Greentea information, please see the [Greentea](../tools/greentea-testing-applications.html) section.
+
+Greentea L3IP tests are in the Mbed OS tree under the `TESTS/network/L3IP` directory.
+
+The driver should also be exercised with real-world examples, such as the [Arm Mbed Client example](https://os.mbed.com/teams/mbed-os-examples/code/mbed-os-example-client/).
```

Comment:
I would prefer to leave this out, and just refer to our Socket test plan.
I have seen it too many times that people skip the socket tests and go directly to test with client. This approach might hide lots of bugs in the socket level, as we have seen.

Reply:
Removed